### PR TITLE
fix: prevent relative links with Markdownlint

### PR DESCRIPTION
**Pull request by nschonni**
### Description

Ported from upstream to ensure relative links aren't used. This one isn't auto-fixable, so I cleaned up the various instances manually in this one

### CODE REVIEW

File: files/zh-tw/web/javascript/reference/operators/this/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ slug: Web/JavaScript/Reference/Operators/this
 
 JavaScript **函式內的 `this` 關鍵字**表現，和其他語言相比略有差異。在[嚴格模式](/zh-TW/docs/Web/JavaScript/Reference/Functions_and_function_scope/Strict_mode)與非嚴格模式下也有所不同。
 
-通常，`this` 值由被呼叫的函式來決定。它不能在執行期間被指派，每次函式呼叫調用的值也可能不同。ES5 引入了 [`bind`](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) 方法去[設置函式的 `this` 值，而不管它怎麼被呼叫。](#The_bind_method)ECMAScript 2015 也導入了定義 `this` 詞法範圍的[箭頭函式](../Functions/Arrow_functions)（它的 `this` 值會維持在詞法作用域）。
+通常，`this` 值由被呼叫的函式來決定。它不能在執行期間被指派，每次函式呼叫調用的值也可能不同。ES5 引入了 [`bind`](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) 方法去[設置函式的 `this` 值，而不管它怎麼被呼叫。](#The_bind_method)ECMAScript 2015 也導入了定義 `this` 詞法範圍的[箭頭函式](/zh-TW/docs/Functions/Arrow_functions)（它的 `this` 值會維持在詞法作用域）。
```
Comment (yin1999) [inline files/zh-tw/web/javascript/reference/operators/this/index.md:10]:
```suggestion
通常，`this` 值由被呼叫的函式來決定。它不能在執行期間被指派，每次函式呼叫調用的值也可能不同。ES5 引入了 [`bind`](/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) 方法去[設置函式的 `this` 值，而不管它怎麼被呼叫。](#The_bind_method)ECMAScript 2015 也導入了定義 `this` 詞法範圍的[箭頭函式](/zh-TW/docs/Web/JavaScript/Reference/Functions/Arrow_functions)（它的 `this` 值會維持在詞法作用域）。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 2]
```diff
@@ -169,7 +169,7 @@ Firefox iOS 版使用 Safari 移动版的默认 User Agent 再加上 **FxiOS/\<v
 - [Firefox OS User Agent](http://lawrencemandel.com/2012/07/27/decision-made-firefox-os-user-agent-string/) （参考博客文章以及 [bug 777710](https://bugzilla.mozilla.org/show_bug.cgi?id=777710)）
 - [Firefox 4 的最终 User Agent](https://hacks.mozilla.org/2010/09/final-user-agent-string-for-firefox-4/)（博客文章）
 - [检测 UA 字符串以获得跨浏览器支持](/zh-CN/Browser_Detection_and_Cross_Browser_Support)
-- [window.navigator.userAgent](../../../../en/DOM/window.navigator.userAgent)
+- [window.navigator.userAgent](/zh-CN/docs/DOM/window.navigator.userAgent)
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:172]:
```suggestion
- [window.navigator.userAgent](/zh-CN/docs/Web/API/Navigator/userAgent)
```

File: files/zh-cn/web/api/node/lookupnamespaceuri/index.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ slug: Web/API/Node/lookupNamespaceURI
 
 返回当前节点上与指定命名空间前缀绑定的命名空间 URI，如果没有，返回`null，如果参数为 null`,返回默认的命名空间。
 
-根据 [bug 312019](https://bugzilla.mozilla.org/show_bug.cgi?id=312019), 该方法对动态指定的命名空间不起作用.(也就是通过[Node.prefix](../../en/DOM/Node.prefix)指定的).
+根据 [bug 312019](https://bugzilla.mozilla.org/show_bug.cgi?id=312019), 该方法对动态指定的命名空间不起作用.(也就是通过[Node.prefix](/zh-CN/docs/DOM/Node.prefix)指定的).
```
Comment (yin1999) [inline files/zh-cn/web/api/node/lookupnamespaceuri/index.md:10]:
```suggestion
根据 [bug 312019](https://bugzilla.mozilla.org/show_bug.cgi?id=312019), 该方法对动态指定的命名空间不起作用。（也就是通过 [Node.prefix](/zh-CN/docs/Web/API/Element/prefix) 指定的）。
```

File: files/zh-cn/web/api/filereadersync/index.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ ArrayBuffer readAsArrayBuffer(
 
 #### 返回值
 
-一个 [`ArrayBuffer`](../JavaScript_typed_arrays/ArrayBuffer) 对象，包含了被读取文件的内容。
+一个 [`ArrayBuffer`](/zh-CN/docs/JavaScript_typed_arrays/ArrayBuffer) 对象，包含了被读取文件的内容。
```
Comment (yin1999) [inline files/zh-cn/web/api/filereadersync/index.md:44]:
```suggestion
一个 [`ArrayBuffer`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) 对象，包含了被读取文件的内容。
```

File: files/zh-cn/web/api/filereadersync/index.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ DOMString readAsDataURL(Blob blob);
 
 ### readAsArrayBuffer()
 
-该方法可以读取指定的 [`Blob`](Blob) 或者 [`File`](File)对象中的内容。当读取完毕后，返回一个 [`ArrayBuffer`](../JavaScript_typed_arrays/ArrayBuffer) 对象，里面包含了被读取文件的内容数据。如果在读取过程中发生了错误，则会抛出相关的异常。
+该方法可以读取指定的 [`Blob`](Blob) 或者 [`File`](File)对象中的内容。当读取完毕后，返回一个 [`ArrayBuffer`](/zh-CN/docs/JavaScript_typed_arrays/ArrayBuffer) 对象，里面包含了被读取文件的内容数据。如果在读取过程中发生了错误，则会抛出相关的异常。
```
Comment (yin1999) [inline files/zh-cn/web/api/filereadersync/index.md:29]:
```suggestion
该方法可以读取指定的 [`Blob`](/zh-CN/docs/Web/API/Blob) 或者 [`File`](/zh-CN/docs/Web/API/File) 对象中的内容。当读取完毕后，返回一个 [`ArrayBuffer`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) 对象，里面包含了被读取文件的内容数据。如果在读取过程中发生了错误，则会抛出相关的异常。
```

File: files/zh-cn/web/api/document/doctype/index.md [hunk 6]
```diff
@@ -3,7 +3,7 @@ title: document.doctype
 slug: Web/API/Document/doctype
 ---
 
-{{ApiRef("DOM")}}返回当前文档关联的文档类型定义 (DTD). 返回的对象实现了 [DocumentType](../../../zh-CN/DOM/DocumentType) 接口。使用 {{domxref("DOMImplementation.createDocumentType()")}} 方法可以创建一个`DocumentType`类型的对象。
+{{ApiRef("DOM")}}返回当前文档关联的文档类型定义 (DTD). 返回的对象实现了 [DocumentType](/zh-CN/DOM/DocumentType) 接口。使用 {{domxref("DOMImplementation.createDocumentType()")}} 方法可以创建一个`DocumentType`类型的对象。
```
Comment (yin1999) [inline files/zh-cn/web/api/document/doctype/index.md:6]:
```suggestion
{{ApiRef("DOM")}}返回当前文档关联的文档类型定义 (DTD). 返回的对象实现了 [DocumentType](/zh-CN/docs/Web/API/DocumentType) 接口。使用 {{domxref("DOMImplementation.createDocumentType()")}} 方法可以创建一个`DocumentType`类型的对象。
```

File: files/zh-cn/web/api/document/beforescriptexecute_event/index.md [hunk 7]
```diff
@@ -7,7 +7,7 @@ slug: Web/API/Document/beforescriptexecute_event
 
 ## 概述
 
-当 HTML 文档中的[`<script>`](../HTML/Element/script)标签内的代码将要执行时触发该事件，如果这个`script`标签是用`appendChild()`等方法动态添加上去的，则不会触发该事件。
+当 HTML 文档中的[`<script>`](/zh-CN/docs/HTML/Element/script)标签内的代码将要执行时触发该事件，如果这个`script`标签是用`appendChild()`等方法动态添加上去的，则不会触发该事件。
```
Comment (yin1999) [inline files/zh-cn/web/api/document/beforescriptexecute_event/index.md:10]:
```suggestion
当 HTML 文档中的[`<script>`](/zh-CN/docs/Web/HTML/Element/script)标签内的代码将要执行时触发该事件，如果这个`script`标签是用`appendChild()`等方法动态添加上去的，则不会触发该事件。
```

File: files/zh-cn/web/api/console/groupcollapsed/index.md [hunk 8]
```diff
@@ -7,7 +7,7 @@ slug: Web/API/Console/groupCollapsed
 
 ## 概述
 
-在 [Web 控制台](../Tools/Web_Console)上创建一个新的分组。随后输出到控制台上的内容都会被添加一个缩进，表示该内容属于当前分组，直到调用[`console.groupEnd()`](../../Article_not_found?uri=zh-cn/DOM/console.groupEnd) 之后，当前分组结束。和 {{ domxref("console.group()") }}方法的不同点是，新建的分组默认是折叠的。用户必须点击一个按钮才能将折叠的内容打开。
+在 [Web 控制台](/zh-CN/docs/Tools/Web_Console)上创建一个新的分组。随后输出到控制台上的内容都会被添加一个缩进，表示该内容属于当前分组，直到调用 {{domxref("console.groupEnd()")}} 之后，当前分组结束。和 {{ domxref("console.group()") }}方法的不同点是，新建的分组默认是折叠的。用户必须点击一个按钮才能将折叠的内容打开。
```
Comment (yin1999) [inline files/zh-cn/web/api/console/groupcollapsed/index.md:10]:
```suggestion
在 [Web 控制台](https://firefox-source-docs.mozilla.org/devtools-user/web_console/index.html)上创建一个新的分组。随后输出到控制台上的内容都会被添加一个缩进，表示该内容属于当前分组，直到调用 {{domxref("console.groupEnd()")}} 之后，当前分组结束。和 {{ domxref("console.group()") }}方法的不同点是，新建的分组默认是折叠的。用户必须点击一个按钮才能将折叠的内容打开。
```

**Review by yin1999 — COMMENTED** [review]
Thanks @nschonni, just some cleaning

**Review by yin1999 — APPROVED** [review]
LGTM for l10n-zh

**Comment by nschonni** [comment]
@yin1999 feel free to push your suggestions next time too, since you actually understand the language 😄

File: files/fr/web/api/window/getselection/index.md [hunk 9]
```diff
@@ -39,8 +39,8 @@ DOM Level 0. Ne fait partie d'aucune spécification.
...
-- [Selection API](../Selection_API)
-- [`Selection`](../Selection)
-- [`Range`](../Range)
-- [`Document.getSelection()`](../Document/getSelection)
-- [`HTMLInputElement.setSelectionRange()`](../HTMLInputElement/setSelectionRange)
+- [Selection API](/fr/docs/Web/API/Selection)
+- {{domxref("Selection")}}
+- {{domxref("Range")}}
+- {{domxref("Document.getSelection()")}}
+- {{domxref("HTMLInputElement.setSelectionRange()")}}
```
Comment (SphinxKnight) [inline files/fr/web/api/window/getselection/index.md:46]:
```suggestion
- [`Selection`](/fr/docs/Web/API/Selection)
- [`Range`](/fr/docs/Web/API/Range)
- [`Document.getSelection()`](/fr/docs/Web/API/Document/getSelection)
- [`HTMLInputElement.setSelectionRange()`](/fr/docs/Web/API/HTMLInputElement/setSelectionRange)
```

Comment (SphinxKnight) [reply files/fr/web/api/window/getselection/index.md]: we try to remove xref macros as we go in `fr`

**Review by SphinxKnight — APPROVED** [review]
Thanks once more @nschonni, I took the freedom of pushing one rewrite to remove macro calls

File: files/fr/web/api/window/getselection/index.md (hunk 9 above)
Comment (nschonni) [reply files/fr/web/api/window/getselection/index.md]: No problem, but I think I copied this from upstream that is still using macros on this  page

File: files/fr/web/api/window/getselection/index.md [hunk 10]
```diff
@@ -39,8 +39,8 @@ DOM Level 0. Ne fait partie d'aucune spécification.
 
 ## Voir aussi
 
-- [Selection API](../Selection_API)
-- [`Selection`](../Selection)
-- [`Range`](../Range)
-- [`Document.getSelection()`](../Document/getSelection)
-- [`HTMLInputElement.setSelectionRange()`](../HTMLInputElement/setSelectionRange)
+- [Selection API](/fr/docs/Web/API/Selection)
+- [`Selection`](/fr/docs/Web/API/Selection)
```
Comment (nschonni) [inline files/fr/web/api/window/getselection/index.md:43]: This link is duplicated now

Comment (SphinxKnight) [reply files/fr/web/api/window/getselection/index.md:43]: 🤦‍♂️ my bad, sorry :/

**Review by SphinxKnight — COMMENTED** [review]
Added a suggestion (and closing a review to be able to push commits) to fix duplicated link

**Review by caugner — APPROVED** [review]

**Review by mfuji09 — APPROVED** [review]
I approve for `ja` document.

**Review by hochan222 — APPROVED** [review]
LGTM for `ko-locale` :)

**Review by josielrocha — APPROVED** [review]
LGTM for `pt-br`.
Thank you @nschonni!

**Comment by SphinxKnight** [comment]
@mdn/yari-content-ru if you have a bit of spare time for this one (as far as I understand it, only `ru`approval is missing)

**Review by sashasushko — APPROVED** [review]
Approve from RU. Thanks 🙏

**Comment by yin1999** [comment]
Approved by all the locales, let's merge this.

**Pull request merged**